docs/style-guide.rst

```diff
@@ -0,0 +1,75 @@
+Style Guide
```

Comment:
It's just a style guide for us for writing the documentation, it's not linked to from anywhere.